### ISSUE TITLE
Vue-3-cypress-vite: Cypress.vue.$on does not exist

### ISSUE THREAD

**Issue by jodyhoonstarr** [cypress-io/cypress#17771]
### Current behavior

Following the `@cypress/vue` [examples](https://www.npmjs.com/package/@cypress/vue) using `$on` to catch an emitted event in a component test gives
```
Cypress.vue.$on is not a function
```
and from TS
```
TS2339: Property '$on' does not exist on type 'ComponentPublicInstance{}, {}, {}, {}, {}, {}, {}, {}, false, ComponentOptionsBase >'.
```

### Desired behavior

Allow a vue3-vite project to catch emitted events in component testing

### Test code to reproduce

Forked from the [vue-3-cypress-vite](https://github.com/lmiller1990/vue-3-cypress-vite) starter repo:
https://github.com/jodyhoonstarr/vue-3-cypress-vite

```
npm i
npm run dev
```
in another terminal
```
npx cypress open-ct
```

### Cypress Version

As of this writing, `npm i` installs
@cypress/vue: 3.0.3
cypress: 7.7.0

**Comment by lmiller1990** [comment]
`$on` is not available in Vue 3. It was [deprecated](https://v3.vuejs.org/guide/migration/events-api.html#overview). 

You can see the examples [here](https://github.com/cypress-io/cypress/tree/develop/npm/vue/examples/vue-cli). `develop` is targeting Vue 3 now.

What are you trying to accomplish? There might be an alternative not using `$on`.

**Comment by JessicaSachs** [comment]
That's because [Vue 3 removed $on](https://v3.vuejs.org/guide/migration/events-api.html#overview). We generally suggest against spying on methods by using Vue so that your tests don't break when Vue breaks the API.

I've written up three different styles of testing your HelloWorld component...
1. Using the component like you would in your application (JSX example)
2. Mounting the component with Vue Test Util's API
3. Mounting the component and using #emitted (recommended approach from Vue Test Utils, however getting the `vueWrapper` inside of Cypress is verbose.

```
import {mount} from '@cypress/vue'
import HelloWorld from './HelloWorld.vue'

// #1 Using the component as you would in your app.
// Equivalent to the below example (JS-only)
// but JSX lets you add styles easily (like resize: both, etc)
describe('JSX Button Counter', () => {
  it('emits an event (with aliases)', () => {
    const spy = cy.spy().as('testEvent')

    mount(() => <HelloWorld onTestEvent={ spy } msg="Hello Cypress!" />)
      .get('#customButton')
      .click()

      // Two ways to make assertions on a spy
      // Option 1: Using the "as" aliass defined on the spy (better DX, prints out named spy in Command Log)
      // Option 2: Asserting on the "spy" directly
      .get('@testEvent').should('have.been.called') 
      .should(() => expect(spy).to.have.been.called) 
  })
})

// #2 Vue compiles methods onto "attrs" in Vue 3.
describe('Using Vue-specific attrs', () => {
  it('emits an event', () => {
    const spy = cy.spy()
    mount(HelloWorld, {
      attrs: { onTestEvent: spy },
      props: { msg: 'Hello Cypress!' }
    })
      .get('#customButton')
      .click()
      .should(() => expect(spy).to.have.been.called)
  })
})


// #3 Using Vue Test Util's #emitted
// How VTU recommends testing emitted events 
// https://next.vue-test-utils.vuejs.org/guide/essentials/event-handling.html#asserting-the-emitted-events
describe('Vue Test Utils Counter', () => {
  beforeEach(() => {
    // propsData => props
    // https://next.vue-test-utils.vuejs.org/migration/#propsdata-is-now-props
    mount(HelloWorld, { props: { msg: 'Hello Cypress!' } })

      // cy.get('@wrapper') returns Vue Test Utils wrapper
      // Cypress.vueWrapper is the result of VTU mounting the component
      // Cypress.vue is the Vue instance that was mounted
      .as('wrapper')
  })

  it('renders a message', () => {
    cy.get('h1').contains('Hello Cypress!')
  })

  it('emits "testEvent" event on click', () => {
    cy.get('#customButton')
      .click()
      .get('@wrapper')

      // Use cy.should instead of cy.then so that your assertions retry
      // https://docs.cypress.io/api/commands/should
      // Doign this solves the kinds of issues you'd have in Vue Test Utils that require $nextTick
      .should(wrapper => {
        expect(wrapper.emitted('testEvent')).to.have.length(1)
      })
  })
})
```

**Comment by hackel** [comment]
I just thought I would mention that using `$on` to spy on events is still mentioned in the [README](https://github.com/cypress-io/cypress/blob/master/npm/vue/README.md) for @cypress/vue.  I realized it was written for vue 2.x, but I assume that was before vue 3 came out.  It would be super helpful to include the examples in https://github.com/cypress-io/cypress/issues/17771#issuecomment-900701935 in the official README.  I realize things are still changing quickly, but there is a lot of old documentation and examples out there.  I landed here after struggling to get `cy.vue()` working to access the wrapper, which I'm assuming was an old API.  Now everything is working great!
